### PR TITLE
feat: add lazy userpreview for all profiles

### LINKED ISSUE #3723: Load popover only on network aka load it from scratch in areas like full profile is there example publications and WTF

### PR DESCRIPTION
**Pull request by Clueed**
### <samp>🤖 Generated by Copilot and @Clueed at ac463fb</samp>

This pull request make UserPreview always fetch from network ([f77ad07](https://github.com/heyxyz/hey/pull/3884/commits/f77ad0744bafb296a57cd6ede3a31b1041d7e3ce)).
Instead, the passed in profile is used as fallback when there is a network error ([ac463fb](https://github.com/heyxyz/hey/pull/3884/commits/ac463fb4c633dc1768e120a56d6bfdae410490a6)).

## Related issues

Fixes #3723

## Type of change

- [x] Enhancement (non-breaking small changes to existing functionality)

### <samp>🤖 Generated by Copilot at ac463fb</samp>

*  Refactor `UserPreview` component to use `useProfileLazyQuery` hook and `lazyProfile` state variable to fetch profile data on hover and improve performance ([link](https://github.com/heyxyz/hey/pull/3884/files?diff=unified&w=0#diff-4cdf48abc8f0279e8ce53beea7dd66d71d57065e549bdf126de7bf715de38567L43-R52))
*  Add optional chaining operators to `isVerified` and `hasMisused` functions to avoid errors when `lazyProfile` or `lazyProfile.id` is undefined ([link](https://github.com/heyxyz/hey/pull/3884/files?diff=unified&w=0#diff-4cdf48abc8f0279e8ce53beea7dd66d71d57065e549bdf126de7bf715de38567L72-R77))
*  Extract `UserProfileContent` component from `UserPreview` to simplify and reuse profile content rendering ([link](https://github.com/heyxyz/hey/pull/3884/files?diff=unified&w=0#diff-4cdf48abc8f0279e8ce53beea7dd66d71d57065e549bdf126de7bf715de38567L87-R98))
*  Replace `lazyProfile` prop with `compositeProfile` prop in `SuperFollow`, `Follow`, and bio rendering to ensure latest profile data is passed and shown ([link](https://github.com/heyxyz/hey/pull/3884/files?diff=unified&w=0#diff-4cdf48abc8f0279e8ce53beea7dd66d71d57065e549bdf126de7bf715de38567L114-R117), [link](https://github.com/heyxyz/hey/pull/3884/files?diff=unified&w=0#diff-4cdf48abc8f0279e8ce53beea7dd66d71d57065e549bdf126de7bf715de38567L126-R123), [link](https://github.com/heyxyz/hey/pull/3884/files?diff=unified&w=0#diff-4cdf48abc8f0279e8ce53beea7dd66d71d57065e549bdf126de7bf715de38567L137-R134), [link](https://github.com/heyxyz/hey/pull/3884/files?diff=unified&w=0#diff-4cdf48abc8f0279e8ce53beea7dd66d71d57065e549bdf126de7bf715de38567L145-R142))
*  Replace `lazyProfile` prop with `compositeProfile` prop in total following and followers count rendering to ensure count is updated with fetched profile data ([link](https://github.com/heyxyz/hey/pull/3884/files?diff=unified&w=0#diff-4cdf48abc8f0279e8ce53beea7dd66d71d57065e549bdf126de7bf715de38567L152-R153), [link](https://github.com/heyxyz/hey/pull/3884/files?diff=unified&w=0#diff-4cdf48abc8f0279e8ce53beea7dd66d71d57065e549bdf126de7bf715de38567L165-R166))
*  Simplify `onPreviewStart` function to use `loadProfile` hook without awaiting or setting loading state and move loading and profile content rendering logic to `Preview` component ([link](https://github.com/heyxyz/hey/pull/3884/files?diff=unified&w=0#diff-4cdf48abc8f0279e8ce53beea7dd66d71d57065e549bdf126de7bf715de38567L182-R203))

## Preview 
https://github.com/heyxyz/hey/assets/7318830/3bdce5eb-12a8-4569-9847-58f04ffae3c4

- Video recoded on slow network to show loading state.
- Only tested on profile as _Who to follow_ only worked when logged-in. Looking at the code it should be no issues.

## Emoji

🚀🧑‍💻🖼️

/claim #3723

### CODE REVIEW

**Comment by Clueed** [comment]
`UserPreview` now strictly only need `handle`. To get rid of the empty profile construction and type assertion in `Mention` in could be refactored to accept `handle` as prop and an optional `fallbackProfile`.    

```ts
  const profile = {
    __typename: 'Profile',
    handle: handle,
    name: null,
    id: null
  };

  return (
    <Link
      href={`/u/${formatHandle(handle)}`}
      onClick={(event) => {
        stopEventPropagation(event);
        Leafwatch.track(PUBLICATION.CLICK_MENTION, {
          handle: formatHandle(handle)
        });
      }}
    >
      {profile?.handle ? (
        <UserPreview
          isBig={false}
          profile={profile as Profile}
```
[link](https://github.com/Clueed/hey/blob/feat/fetch-userpreview/apps/web/src/components/Shared/Markup/MarkupLink/Mention.tsx)


But in all other cases (as far as I can tell), there is always a profile available, which can be undefined makes requiring a handle and leaving the profile optional difficult as they are the same optional objects. 

Here, for example, notification is not typed as optional but always used as `notification?`.
```ts
interface FollowerNotificationProps {
  notification: NewFollowerNotification;
}

const FollowerNotification: FC<FollowerNotificationProps> = ({
  notification
}) => { ...
  return (...
            {notification?.wallet?.defaultProfile ? (
              <UserPreview profile={notification?.wallet?.defaultProfile}>
                <NotificationProfileAvatar
                  profile={notification?.wallet?.defaultProfile}
                />...
```
[link](https://github.com/Clueed/hey/blob/feat/fetch-userpreview/apps/web/src/components/Notification/Type/FollowerNotification.tsx)

Let me know how or if you want to handle this.

**Comment by Clueed** [comment]
I missed some stuff with the fallback implementation. Let me get back you on this. 

It's not necessary to implement the fallback at all but it just makes sense since the data is already there.

File: apps/web/src/components/Shared/UserPreview.tsx [hunk 1]
```diff
@@ -179,26 +176,30 @@ const UserPreview: FC<UserPreviewProps> = ({
...
+  const onPreviewStart = () =>
+    !data &&
+    loadProfile({
+      variables: {
+        request: { handle: formatHandle(profile?.handle, true) }
       }
-
-      setTimeout(() => {
-        setLoading(false);
-      }, 500);
```
Comment (bigint) [inline apps/web/src/components/Shared/UserPreview.tsx:199]: we added 500ms delay intentionally to not load profile instantly 🙇🏼 can we re add back?

Comment (Clueed) [reply apps/web/src/components/Shared/UserPreview.tsx:199]: [Added in last commit.](https://github.com/heyxyz/hey/pull/3884/commits/d09545c636ff38cee9de9798cee6bae58b97d959)

File: apps/web/src/components/Shared/UserPreview.tsx [hunk 2]
```diff
@@ -25,105 +25,140 @@ import Follow from './Profile/Follow';
...
   followStatusLoading,
-  children,
   showUserPreview = true
 }) => {
-  const [lazyProfile, setLazyProfile] = useState(profile);
-  const [loading, setLoading] = useState(false);
-  const [following, setFollowing] = useState(profile?.isFollowedByMe);
+  if (!handle && !profile) {
+    throw new Error('UserPreview needs to receive a profile or a handle');
+  }
```
Comment (bigint) [inline apps/web/src/components/Shared/UserPreview.tsx:51]: We can just return null!

**Review by bigint — CHANGES_REQUESTED** [review]

File: apps/web/src/components/Shared/UserPreview.tsx [hunk 3]
```diff
@@ -26,104 +26,132 @@ import Slug from './Slug';
...
-      className={cn(
-        isBig ? 'h-14 w-14' : 'h-10 w-10',
-        'rounded-full border bg-gray-200 dark:border-gray-700'
-      )}
-      height={isBig ? 56 : 40}
-      width={isBig ? 56 : 40}
-      alt={formatHandle(lazyProfile?.handle)}
-    />
-  );
+  const [fakeLoading, setFakeLoading] = useState<boolean>(loading);
```
[the remark this answers is not in the document]
Comment (Clueed) [reply apps/web/src/components/Shared/UserPreview.tsx]: replaced with `syntheticLoading' and renamed apollo loading to `networkLoading` for clarity.

File: apps/web/src/components/Shared/UserPreview.tsx (hunk 2 above)
Comment (Clueed) [reply apps/web/src/components/Shared/UserPreview.tsx]: Added in last commit.

**Review by bigint — APPROVED** [review]
**LGTM** :shipit: 

LGTM

**Pull request merged**